### DISCUSSION TITLE
P-median with Maximal location simultaneously

### DISCUSSION THREAD

**Discussion by gelazari** [pysal/spaghetti#591] [Q&A]
Using the example with the libraries and neighbourhoods, I came up with a question: is it possible to solve the p-median problem adding the mlcp constraint, i.e. constraint number 6?

I want to solve the p-median problem but taking the maximum distance from facility into account as well . 

All the best,
George

**Comment by jGaboardi** [comment]
Hi, George. I supposed you could try to formulate the problem with these additional constraints, but it would add a lot of complexity and the best would probably be a heuristic if you are actually trying to solve the problem. However, this kind of question is outside of the scope of the `spaghetti` project. I would suggest starting with some of the following articles and investigating from there.



* Chaudhry, S.S., Choi, I. and Smith, D.K. (1995), "Facility location with and without maximum distance constraints through the *p*‐median problem". International Journal of Operations & Production Management, 15(10) 75-81. doi: 10.1108/01443579510098338
* Church, R.L. and Meadows, M.E. (1977), "Results of a New Approach to Solving the *p*‐Median Problem with Maximum Distance Constraints". Geographical Analysis, 9: 364-378. doi: 10.1111/j.1538-4632.1977.tb00589.x
* Khumawala, B.M. (1973), "An Efficient Algorithm for the *p*‐Median Problem With Maximum Distance Constraints". Geographical Analysis, 5: 309-321. doi: 10.1111/j.1538-4632.1973.tb00493.x
* Sáez-Aguado, J., & Trandafir, P. C. (2018), "Variants of the ε-constraint method for biobjective integer programming problems: Application to *p*-median-cover problems". Mathematical Methods of Operations Research, 87(2), 251-283. doi: 10.1007/s00186-017-0618-9

**Reply by gelazari** [reply]
Hi James, thanks for your prompt reply. Isn't there any implementation that I can readily use in Python or somehow incorporate in the code? I thought it would be easier to tackle --perhaps by simply adding constraint 6 to the solver. The reason I'm asking is that I have already created a project based on the libraries/neighbourhoods example. 

Best wishes,
George

**Reply by jGaboardi** [reply]
You can try adding the constraints, but I have no idea if that would work the way you want it to work and I wouldn't be able to help with debugging that. I have no idea if there is Python libraries that can do this, but a search on PyPI/conda-forge/StackExchange might help you. You also might try starting a discussion in [`pysal/spopt`](https://github.com/pysal/spopt/discussions), but I am not sure any of the other maintainers would be able to help further.

**Reply by gelazari** [reply]
How would you approach the problem in order to get a quick solution? I tried adding the constraint but I do not think it has an effect.

**Reply by jGaboardi** [reply]
As I stated in my comment above, I will not be able to assist with this. You're best bet is to search [StackOverflow](https://stackoverflow.com) or maybe [OR](https://or.stackexchange.com). If you start a discussion in [`pysal/spopt`](https://github.com/pysal/spopt/discussions), the other maintainers may be able to help with this (if they have the time/knowledge).
